platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/DraggableMarkerActivity.kt

```diff
@@ -0,0 +1,314 @@
+package com.mapbox.mapboxsdk.testapp.activity.style
+
+import android.graphics.PointF
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v7.app.AppCompatActivity
+import android.view.MotionEvent
+import android.view.View
+import com.mapbox.android.gestures.AndroidGesturesManager
+import com.mapbox.android.gestures.MoveGestureDetector
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.annotations.IconFactory
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.*
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import com.mapbox.mapboxsdk.testapp.R
+import kotlinx.android.synthetic.main.activity_draggable_marker.*
+
+/**
+ * An Activity that showcases how to make symbols draggable.
+ */
+class DraggableMarkerActivity : AppCompatActivity() {
+  companion object {
+    private const val sourceId = "source_draggable"
+    private const val layerId = "layer_draggable"
+    private const val markerImageId = "marker_icon_draggable"
+
+    private var latestId: Long = 0
+    fun generateMarkerId(): String {
+      if (latestId == Long.MAX_VALUE) {
+        throw RuntimeException("You've added too many markers.")
+      }
+      return latestId++.toString()
+    }
+  }
+
+  private val actionBarHeight: Int by lazy {
+    supportActionBar?.height ?: 0
+  }
+
+  private lateinit var mapboxMap: MapboxMap
+  private val featureCollection = FeatureCollection.fromFeatures(mutableListOf())
+  private val source = GeoJsonSource(sourceId, featureCollection)
+  private val layer = SymbolLayer(layerId, sourceId)
+    .withProperties(
+      iconImage(markerImageId),
+      iconAllowOverlap(true),
+      iconIgnorePlacement(true))
+
+  private var draggableSymbolsManager: DraggableSymbolsManager? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_draggable_marker)
+
+    mapView.onCreate(savedInstanceState)
+    mapView.getMapAsync { mapboxMap ->
+      this.mapboxMap = mapboxMap
+
+      // Setting up markers icon, source and layer
+      mapboxMap.addImage(markerImageId, IconFactory.getInstance(this).defaultMarker().bitmap)
+      mapboxMap.addSource(source)
+      mapboxMap.addLayer(layer)
```

Comment:
Steps to replicate the issue.

- Use the API key which has a predefined style with some custom layers created using mapbox studio.

- Apply the style using setStyleUrl()

- Try to add this layer in `onStyleLoaded()` callback.

It was throwing a segmentation fault error when I tried this.

Reply:
The issue was caused because I was trying to add the source before the style has loaded. I tried adding source and layer both in `onStyleLoaded` callback and its working.

Reply:
I'm glad to hear that @iamanvesh! Let us know if you run into any other issues.